woodwork/serializers/parquet_serializer.py

```diff
@@ -26,6 +39,69 @@ def serialize(self, dataframe, profile_name, **kwargs):
         return super().serialize(dataframe, profile_name, **kwargs)
 
     def write_dataframe(self):
-        file = self._get_filename()
-        dataframe = clean_latlong(self.dataframe)
-        dataframe.to_parquet(file, **self.kwargs)
+        # Create a pyarrow table for pandas, skip for Dask/Spark
+        if isinstance(self.dataframe, pd.DataFrame):
+            dataframe = clean_latlong(self.dataframe)
+            self.table = pa.Table.from_pandas(dataframe)
+
+    def write_typing_info(self):
+        loading_info = {
+            "location": self.location,
+            "type": self.format,
+            "params": self.kwargs,
+        }
+        self.typing_info["loading_info"].update(loading_info)
+        # For Dask and Spark we only get the WW metadata because we haven't created
+        # the pyarrow table yet, but for pandas we combine the existing parquet
+        # metadata with the WW metadata.
+        if _is_dask_dataframe(self.dataframe) or _is_spark_dataframe(self.dataframe):
+            metadata = {
+                "ww_meta".encode(): json.dumps(self.typing_info).encode(),
+            }
+        else:
+            table_metadata = self.table.schema.metadata
+            metadata = {
+                "ww_meta".encode(): json.dumps(self.typing_info).encode(),
+                **table_metadata,
+            }
+        self._save_parquet_table_to_disk(metadata)
+
+    def _save_parquet_table_to_disk(self, metadata):
+        if _is_dask_dataframe(self.dataframe):
+            path, dataframe = self._setup_for_dask_and_spark()
+            dataframe.to_parquet(path, custom_metadata=metadata)
+        elif _is_spark_dataframe(self.dataframe):
```

Comment:
This is cool! So with `Dask` and `Pandas`, the writing is fairly straightforward, but for `spark`, we need to write to parquet, open the first file, and then update the metadata of that file with the table schema meta. Interesting how the behaviors of these are so different with parquet support.

Reply:
Yeah, unfortunately these behavior differences make this code flow a lot more complicated than I would prefer, but not sure how to get around it at this point exactly.